### PR TITLE
Updates UnderlineNav to Slots v2

### PR DESCRIPTION
**Pull request by srt32**
Formerly:

```
<%= render(Primer::UnderlineNavComponent.new) do |component| %>
  <% component.with(:body) do %>
    <%= render(Primer::LinkComponent.new(href: "#url")) { "Item 1" } %>
  <% end %>
  <% component.with(:actions) do %>
    <%= render(Primer::ButtonComponent.new) { "Button!" } %>
  <% end %>
<% end %>
```

and now:

```
<%= render(Primer::UnderlineNavComponent.new) do |component| %>
  <% component.body do %>
    <%= render(Primer::LinkComponent.new(href: "#url")) { "Item 1" } %>
  <% end %>
  <% component.actions do %>
    <%= render(Primer::ButtonComponent.new) { "Button!" } %>
  <% end %>
<% end %>
```

### CODE REVIEW

File: app/components/primer/underline_nav_component.rb [hunk 1]
```diff
@@ -5,27 +5,44 @@ module Primer
...
+
+      Primer::BaseComponent.new(tag: :ul, **system_arguments) { content }
+    }
+
+    #  Use actions for a call to action
+    #
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :actions, lambda { |**system_arguments|
+      system_arguments[:tag] ||= :span
+      Primer::BaseComponent.new(**system_arguments) { content }
```
Comment (srt32) [inline app/components/primer/underline_nav_component.rb:27]: This adds a span that we did not have before. Is that okay? Is there a better way to handle this?

File: app/components/primer/underline_nav_component.rb [hunk 2]
```diff
@@ -5,27 +5,44 @@ module Primer
...
+
     ALIGN_DEFAULT = :left
     ALIGN_OPTIONS = [ALIGN_DEFAULT, :right].freeze
 
-    with_content_areas :body, :actions
+    #  Use the body for the navigation items
+    #
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :body, lambda { |**system_arguments|
+      system_arguments[:classes] = class_names("UnderlineNav-body list-style-none", system_arguments[:classes])
```
Comment (manuelpuyol) [inline app/components/primer/underline_nav_component.rb:17]:
```suggestion
      system_arguments[:classes] = class_names("UnderlineNav-body", "list-style-none", system_arguments[:classes])
```

File: app/components/primer/underline_nav_component.rb (hunk 1 above)
Comment (manuelpuyol) [reply app/components/primer/underline_nav_component.rb:27]: looking at https://primer.style/css/components/navigation#underline-nav I feel that we are missing a `<div class="UnderlineNav-actions">`, which could be this slot

**Review by manuelpuyol — COMMENTED** [review]
overall, I don't like how open-ended this component is. It allows user errors (and I think it was rendering the HTML incorrectly before too)

I have two ideas for this component:

1. Nested slots

```rb
class UnderlineNavComponent
  renders_one :body, "Body"
  renders_one :actions, "no changes here"

  class Body
    renders_many :items, ->(href: nil, selected: false) {
      some logic here to choose between "Button" and "Link"
      adding "UnderlineNav-item" class
    }
```

2. In case the body is **never** styled

```rb
class UnderlineNavComponent
  renders_many :items, ->(href: nil, selected: false) {
    some logic here to choose between "Button" and "Link"
    adding "UnderlineNav-item" class
  }
  renders_one :actions, "no changes here"
```

template:

```erb
<div class="UnderlineNav-body">
  <%= items here %>
```

**Comment by manuelpuyol** [comment]
I'd like this component to be as close as possible to `TabNavComponent`

**Comment by srt32** [comment]
I agree there is lots of room for improvement here. I opened https://github.com/primer/view_components/issues/316 to follow up. I propose we leave this component in alpha and we merge this change to SlotsV2 so we can remove all content areas and slots v1 usages and then we can circle back to improve this component

File: app/components/primer/underline_nav_component.rb [hunk 3]
```diff
@@ -5,27 +5,44 @@ module Primer
...
+      system_arguments[:classes] = class_names("UnderlineNav-body", "list-style-none", system_arguments[:classes])
+
+      Primer::BaseComponent.new(tag: :ul, **system_arguments) { content }
+    }
+
+    #  Use actions for a call to action
+    #
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :actions, lambda { |**system_arguments|
+      system_arguments[:tag] ||= :span
```
Comment (srt32) [inline app/components/primer/underline_nav_component.rb:26]:
```suggestion
      system_arguments[:tag] ||= :div
      system_arguments[:classes] = class_names("UnderlineNav-actions", system_arguments[:classes])
```

File: app/components/primer/underline_nav_component.rb [hunk 4]
```diff
@@ -5,27 +5,45 @@ module Primer
...
     ALIGN_OPTIONS = [ALIGN_DEFAULT, :right].freeze
 
-    with_content_areas :body, :actions
+    #  Use the body for the navigation items
+    #
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :body, lambda { |**system_arguments|
+      system_arguments[:classes] = class_names("UnderlineNav-body", "list-style-none", system_arguments[:classes])
+
+      Primer::BaseComponent.new(tag: :ul, **system_arguments) { content }
```
Comment (manuelpuyol) [inline app/components/primer/underline_nav_component.rb:19]: hmm, what happens if someone passes `tag` through  `system_arguments`?
Maybe this is safer:

```suggestion
      system_arguments[:classes] = class_names("UnderlineNav-body", "list-style-none", system_arguments[:classes])
      system_arguments[:tag] = :ul

      Primer::BaseComponent.new(**system_arguments) { content }
```

Comment (srt32) [reply app/components/primer/underline_nav_component.rb:19]: This should be `system_arguments[:tag] ||= :ul`, no?

Comment (manuelpuyol) [reply app/components/primer/underline_nav_component.rb:19]: hmm yeah you're right

Comment (srt32) [inline app/components/primer/underline_nav_component.rb:19]:
```suggestion
      Primer::BaseComponent.new(**system_arguments) { content }
```

**Review by manuelpuyol — APPROVED** [review]

**Pull request merged**